pulp_rpm/tests/functional/api/test_publish.py

```diff
@@ -543,23 +543,20 @@ def do_test(self, mirror):
         from configparser import ConfigParser
 
         # 1. create repo and remote
-        repo = self.repo_api.create(gen_repo())
+        repo = self.repo_api.create(gen_repo(autopublish=not mirror))
         self.addCleanup(self.repo_api.delete, repo.pulp_href)
 
         body = gen_rpm_remote(RPM_KICKSTART_FIXTURE_URL, policy="on_demand")
         remote = self.remote_api.create(body)
         self.addCleanup(self.remote_api.delete, remote.pulp_href)
 
-        # 2. Sync it
+        # 2, 3. Sync and publish
         repository_sync_data = RpmRepositorySyncURL(remote=remote.pulp_href, mirror=mirror)
         sync_response = self.repo_api.sync(repo.pulp_href, repository_sync_data)
-        monitor_task(sync_response.task)
+        created_resources = monitor_task(sync_response.task).created_resources
+
+        publication_href = [r for r in created_resources if "publication" in r][0]
 
-        # 3. Publish and distribute
-        publish_data = RpmRpmPublication(repository=repo.pulp_href)
-        publish_response = self.publications.create(publish_data)
-        created_resources = monitor_task(publish_response.task).created_resources
-        publication_href = created_resources[0]
```

Comment:
Epic fail, this wasn't actually testing the mirrored publication :(  I was doing testing manually also so nothing was missed, and the test wouldn't have caught this, but still.